etcd/cmd/microshift-etcd/run.go

```diff
@@ -95,15 +97,29 @@ func (s *EtcdService) Run() error {
 	if err != nil {
 		return fmt.Errorf("microshift-etcd failed to start: %v", err)
 	}
-	<-e.Server.ReadyNotify()
+	defer func() {
+		e.Server.Stop()
+		<-e.Server.StopNotify()
+	}()
+
+	to := 30 * time.Second
```

Comment:
How was this timeout selected? What happens if it takes longer for etcd to start up? (Say there's a database format change or something and the content has to be migrated.)

This check appears to be happening inside the etcd process, is that right? Why here instead of in the caller?

Reply:
I think we could even drop it, we have a health-check in microshift that checks etcd health (via etcd client)

Reply:
OK, maybe we want to land this PR and then remove the check separately?

Reply:
I just dropped it in a new update

Reply:
Thanks!